scss/_patterns_logo-section.scss

```diff
@@ -0,0 +1,173 @@
+      // When spanning more than one column, we need to account for the gutters between the columns.
+
+      // calc() doesn't allow dividing a mixture of different units, so the calculation
+      // (100% - $gutters-total-width) / $column-count needs to be broken down into two separate divisions:
+      // 100% / $column-count - $gutters-total-width / $column-count
+      // Thus the calculation is split into 2 parts, which are then used in the final calc() function:
+
+      // calculate the width of a single column by dividing the entire width by the number of columns it represents:
+      $percentage-width-of-one-col: 100% / $column-count;
+
+      // multiply the width of one column by the number of columns the logo needs to span
+      $percentage-contribution: $logo-column-span * $percentage-width-of-one-col;
+
+      // calculate the number of gutters the logo spans
+      $gutter-width-inside-logo: $gutter * ($logo-column-span - 1);
+
+      // add the width of gutters spanned by the logo, subtract the $gutters-total-width / $column-count;
+      // the signs here are reversed because the variable is subtracted. This avoids having + and - in the final css, e.g.: "+ -.75rem"
+      $rem-contribution--legacy-browsers: -$gutter-width-inside-logo;
+      $rem-contribution: $logo-column-span * ($gutters-total-width / $column-count) - $gutter-width-inside-logo;
+
+      .#{$grid-column-prefix}#{$column-count} & {
+        @include vf-p-apply-logo-section-item-width($percentage-contribution, $rem-contribution--legacy-browsers);
+
+        @supports (gap: 1rem) {
+          @include vf-p-apply-logo-section-item-width($percentage-contribution, $rem-contribution);
+        }
+      }
+    }
+  }
+}
+
+@mixin vf-p-apply-logo-section-item-width($percentage-contribution, $rem-contribution) {
+  @if $rem-contribution == 0 {
+    width: $percentage-contribution;
+  } @else {
+    width: calc(#{$percentage-contribution} - #{$rem-contribution});
+  }
+}
+
+@mixin vf-p-logo-section {
+  // The logo section pattern needs to align logos to the grid when they are an even number, and center them (while maintaining the same logo width) when the number of logos on a line is odd.
+  // This means we can't use css grid for this, instead we need to replicate its behavior for even numbers of logos, and center the odd ones.
+  .p-logo-section__title {
+    @extend %x-small-text;
+
+    font-weight: $font-weight-bold;
+    text-transform: uppercase;
+  }
+
+  .p-logo-section__items {
+    display: flex;
+    flex-wrap: wrap;
+    gap: 0 map-get($grid-gutter-widths, small);
+    justify-content: center;
+    margin-bottom: $spv-outer--scaleable;
+    width: 100%;
+
+    @media (min-width: $threshold-4-6-col) {
+      @supports (gap: 1rem) {
+        gap: 0 map-get($grid-gutter-widths, medium);
+      }
+    }
+
+    @media (min-width: $threshold-6-12-col) {
+      @supports (gap: 1rem) {
```

Comment:
Do we need `@supports` here? If we only use `gap` inside of it, it will be ignored by browsers not supporting anyway.